tests/benchmarks/test_plugin_render.py

```
@@ -0,0 +1,229 @@
+        result = benchmark(weather.generate_image, settings, device_cfg)
+
+    assert isinstance(result, Image.Image)
+
+
+# ---------------------------------------------------------------------------
+# 3. HTML→PIL pipeline (base plugin render_image via plugin.html)
+# ---------------------------------------------------------------------------
+
+
+@pytest.mark.benchmark(group="plugin_render")
+def test_bench_html_render(benchmark, device_cfg):
```

Ruff (0.15.9)

[warning] Unused function argument: `device_cfg`

(ARG001)